src/raft/log.c

```diff
@@ -243,62 +298,70 @@ static int refsInit(struct raft_log *l,
 	 * where this is not enough. */
 	for (i = 0; i < 10; i++) {
 		bool collision;
+		struct raft_entry_ref *ref;
 		int rc;
 
-		rc = refsTryInsert(l->refs, l->refs_size, term, index, 1, buf,
-				   batch, &collision);
-		if (rc != 0) {
-			return RAFT_NOMEM;
-		}
-
-		if (!collision) {
-			return 0;
+		ref = refsTryInsert(l->refs, l->refs_size, term, index, 1, buf,
+				    batch, &collision);
+		if (ref == NULL && !collision) {
+			return NULL;
```

Comment:
Initial logic looks a bit different. `!collision` looks to be a new element in this case.

Reply:
Fixed, the two return values are unnecessarily hard to reason about so I switched to using a non-NULL sentinel for the collision case, which allows hewing more closely to the original code structure.